dasharo-compatibility/boot-maintenance-manager.robot

```diff
@@ -0,0 +1,73 @@
+*** Settings ***
+Library             Collections
+Library             OperatingSystem
+Library             Process
+Library             String
+Library             Telnet    timeout=20 seconds    connection_timeout=120 seconds
+Library             SSHLibrary    timeout=90 seconds
+Library             RequestsLibrary
+# TODO: maybe have a single file to include if we need to include the same
+# stuff in all test cases
+Resource            ../sonoff-rest-api/sonoff-api.robot
+Resource            ../rtectrl-rest-api/rtectrl.robot
+Resource            ../variables.robot
+Resource            ../keywords.robot
+Resource            ../keys.robot
+
+# TODO:
+# - document which setup/teardown keywords to use and what are they doing
+# - go threough them and make sure they are doing what the name suggest (not
+# exactly the case right now)
+Suite Setup         Run Keyword
+...                     Prepare Test Suite
+Suite Teardown      Run Keyword
+...                     Log Out And Close Connection
+
+
+*** Test Cases ***
+BMM001.001 Set Auto Boot Time-out to 7 and check after reboot
+    [Documentation]    Check whether setting Auto Boot Time-out to 7 the value
+    ...    is remembered after restart
+    Skip If    not ${RESET_TO_DEFAULTS_SUPPORT}
+    Skip If    not ${TESTS_IN_FIRMWARE_SUPPORT}    RTD011.001 not supported
+    Power On
+    Enter Setup Menu Tianocore
+    Enter Dasharo Submenu    Boot Maintenance Manager
+    Refresh Serial Screen In BIOS Editable Settings Menu
+    Change Numeric Value Of Setting    Auto Boot Time-out    7    Esc=Exit
+    Press Key N Times    1    ${F10}
+    Write Bare Into Terminal    y
+    Read From Terminal Until    Esc=Exit
+    Restart The DUT
+    Enter Setup Menu Tianocore
+    Enter Dasharo Submenu    Boot Maintenance Manager
+    Refresh Serial Screen In BIOS Editable Settings Menu
+    ${value}=    Get Option Value    Auto Boot Time-out    Esc=Exit
+    Should Be Equal    ${value}    [7]
+
+BMM002.001 F9 resets Auto Boot Time-out to default value
+    [Documentation]    Check whether pressing F9 resets Auto Boot Time-out to
+    ...    default value
+    Skip If    not ${RESET_TO_DEFAULTS_SUPPORT}
+    Skip If    not ${TESTS_IN_FIRMWARE_SUPPORT}    RTD011.001 not supported
+    Power On
+    Enter Setup Menu Tianocore
+    Enter Dasharo Submenu    Boot Maintenance Manager
+    Refresh Serial Screen In BIOS Editable Settings Menu
+    Reset To Defaults Tianocore    Esc=Exit
+    Press Key N Times    1    ${F10}
+    Write Bare Into Terminal    y
+    Read From Terminal Until    Esc=Exit
+    Restart The DUT
+    Enter Setup Menu Tianocore
+    Enter Dasharo Submenu    Boot Maintenance Manager
+    Refresh Serial Screen In BIOS Editable Settings Menu
+    ${value}=    Get Option Value    Auto Boot Time-out    Esc=Exit
+    Should Be Equal    ${value}    ${AUTO_BOOT_TIME_OUT_DEFAULT_VALUE}
```

Comment:
@Psotas I wonder how this could have worked, if the `[6]` is being compared with `6` from config. In the above test you specifically compared it with `[7]`. So I just suspect it has not worked.

Reply:
The first test aims to change the value to an arbitrary one and compare if the change was applied.

Second test check if timeout resets back to the default, which is hardcoded in config.

I don't see any error with that approach @macpijan except the square brackets which should be rather stripped by BIOS menu keywords

Reply:
The test with improved keywords is moved here: https://github.com/Dasharo/open-source-firmware-validation/pull/55/files#diff-3c157e9d4762250f95224f014fa3acb1f979e1aca0c9fe3808a4156f980d7cc7R28

Reply:
> I don't see any error with that approach @macpijan except the square brackets which should be rather stripped by BIOS menu keywords 

I was just only saying that it couldn't have worked, since one test case assume they are stripped, the other assumes otherwise. And @Psotas claimed the both worked.

Reply:
> Please do not waste time on this MR. 

Unfortunately this won't always work in my workmodel. I go through each of the notifications in order. So, depending on what notifications I go through first (whether mail or matrix), I may, or may not, notice such announcements.

Reply:
If it would be closed (as I asked) you would not have reviewed close MR I assume.